tracer/src/Datadog.Trace/Util/Http/HttpRequestUtils.cs

```diff
@@ -34,6 +36,6 @@ internal static string GetUrl(string scheme, string host, int? port, string path
             return $"{scheme}://{GetNormalizedHost(host)}{(port.HasValue ? $":{port}" : string.Empty)}{pathBase}{path}";
         }
 
-        internal static string GetNormalizedHost(string host) => string.IsNullOrEmpty(host) ? NoHostSpecified : host;
+        internal static string GetNormalizedHost(string? host) => host is null || host.Length == 0 ? NoHostSpecified : host;
```

Comment:
Thanks for updating the nullability requirements here 👍 

IsNullOrEmpty is optimised so is generally preferred - to fix the nullability issue we have a helper, `StringUtil` 🙂 

```suggestion
        internal static string GetNormalizedHost(string? host) => StringUtil.IsNullOrEmpty(host) ? NoHostSpecified : host;
```